pkg/controller/kubelet-config/helpers.go

```diff
@@ -110,13 +133,87 @@ func updateOriginalKubeConfigwithNodeConfig(node *osev1.Node, originalKubeletCon
 		originalKubeletConfig.NodeStatusUpdateFrequency = metav1.Duration{Duration: osev1.LowNodeStatusUpdateFrequency}
 	case osev1.DefaultUpdateDefaultReaction:
 		originalKubeletConfig.NodeStatusUpdateFrequency = metav1.Duration{Duration: osev1.DefaultNodeStatusUpdateFrequency}
+	case emptyInput:
+		return nil
 	default:
 		return fmt.Errorf("unknown worker latency profile type found %v, failed to update the original kubelet configuration", node.Spec.WorkerLatencyProfile)
 	}
 	// The kubelet configuration can be updated based on the cgroupmode as well here.
 	return nil
 }
 
+// isTechPreviewNoUpgradeEnabled returns a boolean accordingly if a feature is enabled
+// (TODO) This can be generically used in future by passing any feature as an argument.
+func isTechPreviewNoUpgradeEnabled(features *osev1.FeatureGate) bool {
+	if features == nil {
+		return false
+	}
+	if features.Spec.FeatureGateSelection.FeatureSet == osev1.TechPreviewNoUpgrade {
+		featureSet, ok := osev1.FeatureSets[osev1.TechPreviewNoUpgrade]
+		if !ok {
+			return false
+		}
+		for _, val := range featureSet.Enabled {
+			if val == cgroupv2Feature {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// updateMachineConfigwithCgroup updates the Machine Config object based on the cgroup mode present in the Config Node resource.
+func updateMachineConfigwithCgroup(node *osev1.Node, mc *mcfgv1.MachineConfig) error {
+	if node == nil {
+		return fmt.Errorf("node configuration not found, failed to update the machine config with the cgroup information")
+	}
+	if reflect.DeepEqual(node.Spec, osev1.NodeSpec{}) {
+		return fmt.Errorf("empty config node resource spec found")
+	}
+	if mc == nil || reflect.DeepEqual(mc.Spec, mcfgv1.MachineConfigSpec{}) {
+		return fmt.Errorf("machine config not found, failed to update the machine config with the cgroup information")
+	}
+	// updating the Machine Config resource with the relevant cgroup config
+	var (
+		kernelArgsv1       = []string{"systemd.unified_cgroup_hierarchy=0", "systemd.legacy_systemd_cgroup_controller=1"}
+		kernelArgsv2       = []string{"systemd.unified_cgroup_hierarchy=1", "cgroup_no_v1=\"all\"", "psi=1"}
+		kernelArgstoRemove []string
+		kernelArgstoAdd    []string
+	)
+	switch node.Spec.CgroupMode {
+	case osev1.CgroupModeV1:
+		kernelArgstoAdd = append(kernelArgstoAdd, kernelArgsv1...)
+		kernelArgstoRemove = append(kernelArgstoRemove, kernelArgsv2...)
+	case osev1.CgroupModeV2:
+		kernelArgstoAdd = append(kernelArgstoAdd, kernelArgsv2...)
+		kernelArgstoRemove = append(kernelArgstoRemove, kernelArgsv1...)
+	case emptyInput:
```

Comment:
emptyInput should be to use cgroupv1

Reply:
By default, the node spec is empty when a cluster is up and hence the cgroupMode is an empty string.
If the emptyInput is to be interpreted as "v1", there will be an unnecessary MC for each pool created and so the MCP starts getting updated(default system comes up with "v1" anyway)
Hence we intend to update the cluster only when a user explicitly gives the appropriate input value.

Reply:
No, it is not reverted back to the default behaviour.